aten/src/ATen/Context.cpp

```diff
@@ -205,32 +261,69 @@ void Context::setBenchmarkLimitCuDNN(int b) {
 }
 
 bool Context::allowTF32CuBLAS() const {
-  return float32_matmul_precision != at::Float32MatmulPrecision::HIGHEST;
+  bool legacy_allow_tf32 = float32_matmul_precision != at::Float32MatmulPrecision::HIGHEST;
+  bool allow_tf32 = float32Precision("cuda", "matmul") == "tf32" ? true : false;
+  TORCH_CHECK(
+      legacy_allow_tf32 == allow_tf32,
+      "You are checking whether allow_tf32 is enabled for cublas matmul. ",
+      "From Pytorch 2.4 we introduced an new set of API to set the TF32 status for cublas matmul. ",
+      "Current status indicate that you have used mix of legacy API and new API to set the TF32 status for cublas matmul. ",
+      "We suggest to only use the new API to set the TF32 status, you can check the usage at ",
+      "https://github.com/pytorch/pytorch/blob/main/docs/source/notes/cuda.rst#tensorfloat-32-tf32-on-ampere-and-later-devices");
```

Comment:
Why does the legacy variable need to be maintained here? Would it be possible to just convert medium/high -> allow tf32, and highest -> don't allow tf32 when that codepath is used?

Reply:
This is similar with the cudnn allow tf32.
Legacy API matmul_float32_precision control both mkldnn matmul and cublas matmul. 
I leave the legacy variable here to know whether users use both legacy and new APIs.
If user use new API to set mkldnn matmul to bf16, we actually do not know what user should get if he query with legacy API `torch.matmul_float32_precision`. We maintained two variables here to detect whether use are use mix of both APIs, and we can deprecate legacy APIs after 2 version.